xapian-letor/api/letor.cc

```diff
@@ -36,8 +36,22 @@ namespace Xapian {
 
 Letor::Letor() : internal(new Letor::Internal) { }
 
-Letor::~Letor() { }
+Letor::Letor(const Xapian::Database & db, Xapian::Ranker * ranker) {
+    internal = new Letor::Internal();
+    internal->letor_db = db;
+    internal->ranker = ranker;
+}
 
+Letor::Letor(const Xapian::Database & db, const Xapian::Query & query, Xapian::Ranker * ranker) {
+    internal = new Letor::Internal();
+    internal->letor_db = db;
+    internal->letor_query = query;
+    internal->ranker = ranker;
+}
+
+Letor::~Letor() {
+    delete internal->ranker;
```

Comment:
If you're going to do this, it should be documented in the API to avoid people getting confused as to why the `Ranker` they made has disappeared. (Unless it's possible to use the ref counting stuff here?)